### DISCUSSION TITLE
[General] dubbo 2.7.22 still has broken connection issue

### DISCUSSION THREAD

**Discussion by zrlw** [apache/dubbo#14189] [General]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

we have a dubbo 2.7.22 consumer A that will call  dubbo 2.7.22 provider B only if  it received request from dubbo 2.7.22 consumer C:
```
C ---RPC---> A ---RPC---> B
```
C might call A once in a few days or weeks, so the dubbo connection between A and B is rarely used.
recently we found that the result of A calling B is always be no provider exception, and the dubbo connections between A and B is gone. But the ip address of A is still at the consumer list of provider B on the registry center (Nacos 1.4.2). 
we tested the network by telnet from A to B dubbo service port, the network is OK. 
we checked ReferenceCountExchangeClient.java and found it's already be patched by #7322 ```https://github.com/apache/dubbo/pull/7322```
anyone has any ideas about how to deal with it will be appreciated.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by aofall** [comment]
Upgrade versions is priority. If you are concerned issue caused by the upgrade, you can also edit and repackage from source code by yourself.

Here are the other way to replace the class from the jar. Copy the source code and directly create the same package with the same name class replacement in the project, edit to patched the issue, but you must ensure that its package name and class name is the same as the original class, the Spring project will prioritize the loading of customized classes to replace the class in the jar package, the class is applicable to the Spring project to replace
